firestore/test.solution-geoqueries.js

```diff
@@ -0,0 +1,103 @@
+import firebase from 'firebase/app';
+import 'firebase/firestore';
+
+const geofire = require('geofire-common');
+
+/**
+ * @type firebase.firestore.Firestore
+ */
+var db;
+
+function addHash(done) {
+  // [START fs_geo_add_hash]
+
+  // Compute the GeoHash for a lat/lng point
+  const lat = 51.5074;
+  const lng = 0.1278;
+  const hash = geofire.geohashForLocation([lat, lng]);
+
+  // Add the hasn and the lat/lng to the document. We will use the hash
+  // for queries and the lat/lng for distance comparisons.
+  const londonRef = db.collection('cities').doc('LON');
+  londonRef.update({
```

Comment:
Any specific reason to not use a `GeoPoint` here?

See https://stackblitz.com/edit/geofire-common?file=index.js

Reply:
I thought about it but honestly a GeoPoint is negative value today.  They just make your data less portable and reduce your ability to query on `lat` or `lng` individually (if you ever needed to do that).

I can't think of a situation where they help you, they mostly hurt you and they're neutral at best.